packages/jest-matchers/src/__tests__/__snapshots__/spyMatchers-test.js.snap

```diff
@@ -1,27 +1,444 @@
+exports[`test lastCalledWith works with jest.fn and many arguments 1`] = `
+[Error: [2mexpect([22m[31mjest.fn()[39m[2m).not.lastCalledWith([22m[32mexpected[39m[2m)[22m
 
-exports[`test toHaveBeenCalled works only on spies or jest.fn 1`] = `[Error: toHaveBeenCalled matcher can only be used on a spy or mock function.]`;
+Expected mock function to not have been last called with:
+  [32m["foo","bar"][39m]
+`;
 
-exports[`test toHaveBeenCalled works with jasmine.createSpy 1`] = `[Error: Expected the spy to be called.]`;
+exports[`test lastCalledWith works with jest.fn and many arguments that don't match 1`] = `
+[Error: [2mexpect([22m[31mjest.fn()[39m[2m).lastCalledWith([22m[32mexpected[39m[2m)[22m
 
-exports[`test toHaveBeenCalled works with jasmine.createSpy 2`] = `[Error: Expected the spy not to be called, but it was called 1 time.]`;
+Expected mock function to have been last called with:
+  [32m["foo","bar"][39m
+But it was last called with:
+  [31m["foo","bar3"][39m
+and [31mtwo more calls[39m.]
+`;
 
-exports[`test toHaveBeenCalled works with jasmine.createSpy 3`] = `[Error: toHaveBeenCalled matcher does not accept any arguments.]`;
+exports[`test lastCalledWith works with jest.fn and no arguments 1`] = `
+[Error: [2mexpect([22m[31mjest.fn()[39m[2m).lastCalledWith([22m[32mexpected[39m[2m)[22m
 
-exports[`test toHaveBeenCalled works with jest.fn 1`] = `[Error: Expected the mock function to be called.]`;
+Expected mock function to have been last called with:
+  [32m["foo","bar"][39m
+But it was [31mnot called[39m.]
+`;
 
-exports[`test toHaveBeenCalled works with jest.fn 2`] = `[Error: Expected the mock function not to be called, but it was called 1 time.]`;
+exports[`test toBeCalled works only on spies or jest.fn 1`] = `
+[Error: [2mexpect([22m[31mjest.fn()[39m[2m)[.not].toBeCalled([22m[2m)[22m
 
-exports[`test toHaveBeenCalled works with jest.fn 3`] = `[Error: toHaveBeenCalled matcher does not accept any arguments.]`;
+[31mjest.fn()[39m value must be a mock function or spy.
+Received:
+  function: [31m"() => {}"[39m]
+`;
+
+exports[`test toBeCalled works with jasmine.createSpy 1`] = `
+[Error: [2mexpect([22m[31mspy[39m[2m).toBeCalled([22m[2m)[22m
+
+Expected spy to have been called.]
+`;
+
+exports[`test toBeCalled works with jasmine.createSpy 2`] = `
+[Error: [2mexpect([22m[31mspy[39m[2m).not.toBeCalled([22m[2m)[22m
+
+Expected spy not to be called, but it was called [31mone time[39m.]
+`;
+
+exports[`test toBeCalled works with jasmine.createSpy 3`] = `
+[Error: [2mexpect([22m[31mreceived[39m[2m)[.not].toBeCalled([22m[2m)[22m
+
+Matcher does not accept any arguments.
+Got:
+  number: [32m555[39m]
+`;
+
+exports[`test toBeCalled works with jest.fn 1`] = `
+[Error: [2mexpect([22m[31mjest.fn()[39m[2m).toBeCalled([22m[2m)[22m
+
+Expected mock function to have been called.]
```

Comment:
We had this before but it was pretty odd. I think it is quite obvious that if we expected something to have happened, it didn't actually happen without printing the same thing twice.